cishouseholds/pipeline/high_level_transformations.py

```diff
@@ -2643,3 +2647,292 @@ def flag_records_to_reclassify(df: DataFrame) -> DataFrame:
+        negative_regex_pattern=not_working_pattern.negative_regex_pattern,
+    )
+
+    update_work_status_not_working_v0 = not_working_regex_hit & flag_records_for_not_working_rules_v0()
+    update_work_status_not_working_v1_a = not_working_regex_hit & flag_records_for_not_working_rules_v1_a()
+    update_work_status_not_working_v1_b = not_working_regex_hit & flag_records_for_not_working_rules_v1_b()
+    update_work_status_not_working_v2_a = not_working_regex_hit & flag_records_for_not_working_rules_v2_a()
+    update_work_status_not_working_v2_b = not_working_regex_hit & flag_records_for_not_working_rules_v2_b()
+
+    # School/Student
+    school_regex_hit = regex_match_result(
+        columns_to_check_in=["work_main_job_title", "work_main_job_role"],
+        positive_regex_pattern=at_school_pattern.positive_regex_pattern,
+        negative_regex_pattern=at_school_pattern.negative_regex_pattern,
+    )
+
+    college_regex_hit = regex_match_result(
+        columns_to_check_in=["work_main_job_title", "work_main_job_role"],
+        positive_regex_pattern=in_college_or_further_education_pattern.positive_regex_pattern,
+        negative_regex_pattern=in_college_or_further_education_pattern.negative_regex_pattern,
+    )
+
+    university_regex_hit = regex_match_result(
+        columns_to_check_in=["work_main_job_title", "work_main_job_role"],
+        positive_regex_pattern=at_university_pattern.positive_regex_pattern,
+        negative_regex_pattern=at_university_pattern.negative_regex_pattern,
+    )
+    under_16 = F.col("age_at_visit") < F.lit(16)
+
+    update_work_status_student_v0 = (
+        (school_regex_hit & flag_records_for_student_v0_rules())
+        | (university_regex_hit & flag_records_for_student_v0_rules())
+        | (college_regex_hit & flag_records_for_student_v0_rules())
+        | under_16
+    )
+
+    update_work_status_student_v1 = (
+        (school_regex_hit & flag_records_for_student_v1_rules())
+        | (university_regex_hit & flag_records_for_student_v1_rules())
+        | (college_regex_hit & flag_records_for_student_v1_rules())
+        | under_16
+    )
+
+    update_work_status_student_v2_a = (school_regex_hit & flag_records_for_student_v2_rules()) | under_16
+
+    update_work_status_student_v2_b = college_regex_hit & flag_records_for_college_v2_rules()
+
+    update_work_status_student_v2_c = university_regex_hit & flag_records_for_uni_v2_rules()
+
+    update_work_location_general = F.col("work_location").isNull() & (
+        F.col("work_status_v0").isin(
+            "Furloughed (temporarily not working)",
+            "Not working (unemployed, retired, long-term sick etc.)",
+            "Student",
+        )
+    )
+
+    # Please note the order of *_edited columns, these must come before the in-place updates
+
+    # first start by taking a copy of the original work variables
+    _df = (
+        df.withColumn("work_location_original", F.col("work_location"))
```

Comment:
My only thought is that we already create "raw_" versions of some of these in `transform_survey_responses_generic`. So it might be that we can use these here instead? That is assuming that the non-raw versions don't get edited somewhere else in the code before this point